partiql-parser/src/main/antlr/PartiQLParser.g4

```diff
@@ -623,8 +623,8 @@ exprPrimary
     | substring                  # ExprPrimaryBase
     | position                   # ExprPrimaryBase
     | overlay                    # ExprPrimaryBase
-    | canCast                    # ExprPrimaryBase
-    | canLosslessCast            # ExprPrimaryBase
+    | canCast                    # ExprPrimaryBase  // TODO remove ahead of `v1` release
```

Comment:
(self-review) Was not sure if we wanted to remove the `partiql-parser` logic just yet or wait until after we've deprecated everything in `partiql-lang` before removing.

Reply:
from offline discussion, we'll keep the parsing rules in the grammar in place for now. Eventually, the rules in the parser and lexer will be removed in the `v1` branch. We will have the following behavior for the ANTLR visitors:
- partiql-parser ANTLR visitor (parsing to partiql-ast) -- will now error when visiting the `canCast` and `canLosslessCast` nodes
- partiql-lang ANTLR visitor (parsing to PIG-generated AST) -- leave as-is